lib/restaurant.rb

```diff
@@ -0,0 +1,63 @@
+require_relative 'menu'
+require 'twilio-ruby'
+#require 'yaml'
+
+class Restaurant
+
+  attr_accessor :order, :menu
+
+  def initialize(menu_class = Menu)
+    @order = []
+    @menu = menu_class.new
+  end
+
+  def display_menu
+  @menu.show_menu.each do |k|
+     "#{k}"
+   end
+  end
+
+  def add_to_order(dish, quantity = 1)
+    dish = dish.capitalize
+    fail "#{dish} isn't available" if dish_check(dish)
+     add(dish, quantity)
+  end
+
+  def cost
+    @cost = @order.inject(0) {|sum, hash| sum + hash[:price]}
+    "This order will cost £#{@cost}0"
+  end
+
+  def confirm_order(cost)
+   fail 'Please enter the correct amount' unless @cost == cost
```

Comment:
This method relies on a previous call to `cost` to initialize `@cost` which will cause inconsistent behaviour.